### PR TITLE
Edit an AWS Account

### PR DESCRIPTION
**Pull request by thibautcornolti**
Added PATCH method to the /aws route to edit an AWS Account. Only the pretty field can be edited.

### CODE REVIEW

File: aws/aws.go [hunk 1]
```diff
@@ -137,6 +137,20 @@ func (a *AwsAccount) CreateAwsAccount(ctx context.Context, db models.XODB) error
 	return err
 }
 
+// UpdateAWSAccount updates an AWS account for a user. It does no error
+// checking: the caller should check themselves that the AWS account exists.
+// Only the Pretty will be updated.
+func (a *AwsAccount) UpdateAWSAccount(ctx context.Context, tx *sql.Tx) error {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	dbAwsAccount, _ := models.AwsAccountByID(tx, a.Id)
```
Comment (giubil) [inline aws/aws.go]: Why don't you check the error return value of `models.AwsAccountByID()` ?

File: aws/routeAwsPatch.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+// patchAwsAccountWithValidBody handles the logic of patchAwsAccount assuming the
+// request body is valid.
+func patchAwsAccountWithValidBody(r *http.Request, tx *sql.Tx, user users.User, body patchAwsAccountRequestBody, id int) (int, interface{}) {
+	ctx := r.Context()
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	awsAccount, err := GetAwsAccountWithIdFromUser(user, id, tx)
+	if err == nil {
+		awsAccount.Pretty = body.Pretty
+		if err := awsAccount.UpdateAWSAccount(ctx, tx); err != nil {
+			return 500, errFailUpdateAccount
```
Comment (giubil) [inline aws/routeAwsPatch.go]: You should log an error here

**Review by giubil — CHANGES_REQUESTED** [review]

**Review by giubil — DISMISSED** [review]

File: aws/aws.go [hunk 3]
```diff
@@ -137,6 +137,24 @@ func (a *AwsAccount) CreateAwsAccount(ctx context.Context, db models.XODB) error
 	return err
 }
 
+// UpdateAWSAccount updates an AWS account for a user. It does no error
+// checking: the caller should check themselves that the AWS account exists.
+// Only the Pretty will be updated.
+func (a *AwsAccount) UpdateAWSAccount(ctx context.Context, tx *sql.Tx) error {
```
Comment (schubev) [inline aws/aws.go]: If only the `pretty` is updated, the function’s name should reflect it.

File: aws/aws.go [hunk 4]
```diff
@@ -137,6 +137,24 @@ func (a *AwsAccount) CreateAwsAccount(ctx context.Context, db models.XODB) error
...
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	dbAwsAccount, err := models.AwsAccountByID(tx, a.Id)
+	if err != nil {
+		logger.Error("Failed to get AWS account in database.", nil)
+	} else {
+		dbAwsAccount.Pretty = a.Pretty
+		err := dbAwsAccount.Update(tx)
```
Comment (schubev) [inline aws/aws.go]: By using `:=` here you shadow the `err` defined at line 145, so at line 155 you would still return `nil`.

Comment (schubev) [reply aws/aws.go]: Please write unit tests, you should have caught this with unit tests.

File: aws/routes.go [hunk 5]
```diff
@@ -27,6 +27,13 @@ import (
 	"github.com/trackit/trackit2/users"
 )
 
+var (
+	// QueryArgAwsAccount allows to get the AWS Account ID in the URL Parameters
+	// with routes.RequiredQueryArgs. This AWS Account ID will be an Uint stored in
+	// the routes.Arguments map with itself for key.
+	QueryArgAwsAccount = routes.QueryArg{"aa", "AWS Account ID", routes.QueryArgUint{}}
```
Comment (schubev) [inline aws/routes.go]: Isn’t that already defined somewhere else ?

**Review by schubev — DISMISSED** [review]
Beware of variable shadowing.

File: aws/aws.go (hunk 4 above)
Comment (thibautcornolti) [reply aws/aws.go]: `err` is shadowed only if the `err` defined at line 145 is `nil`. It's the expected behaviour.

File: aws/aws.go [hunk 6]
```diff
@@ -137,6 +137,24 @@ func (a *AwsAccount) CreateAwsAccount(ctx context.Context, db models.XODB) error
 	return err
 }
 
+// UpdatePrettyAWSAccount updates an AWS account for a user. It does no error
+// checking: the caller should check themselves that the AWS account exists.
+// Only the Pretty will be updated.
+func (a *AwsAccount) UpdatePrettyAWSAccount(ctx context.Context, tx *sql.Tx) error {
```
Comment (schubev) [inline aws/aws.go]: The convention in this project is to treat acronyms as simple words, with only the first letter capitalized, so this should be `UpdatePrettyAwsAccount`.

**Review by schubev — CHANGES_REQUESTED** [review]
Just one change to an identifier, and you’re good !

**Review by schubev — APPROVED** [review]

**Pull request merged**